cmd/snap/main.go

```diff
@@ -158,6 +161,17 @@ func init() {
 
 func main() {
 	cmd.ExecInCoreSnap()
+
+	// magic \o/
+	snapApp := filepath.Base(os.Args[0])
+	if osutil.IsSymlink(filepath.Join(dirs.SnapBinariesDir, snapApp)) {
+		cmd := &cmdRun{}
+		cmd.Positional.SnapApp = snapApp
+		cmd.Execute(os.Args[1:])
+		return
```

Comment:
so, um.
I know I *said* return. But what you actually want is is to `os.Exit` with either 1 or 0 depending on whether `cmd.Execute` returns an error or not (or maybe: 0 if err is nil, int(err.(syscall.Error)) if it's a syscall.Error, 1 otherwise?)